client/src/app/pages/assessment/components/assessment-wizard/assessment-wizard.tsx

```diff
@@ -610,22 +551,81 @@ export const AssessmentWizard: React.FC<AssessmentWizardProps> = ({
     }
   };
 
+  const getWizardFooter = (step: number, section?: Section) => {
+    return (
+      <CustomWizardFooter
+        enableNext={
+          section ? shouldNextBtnBeEnabled(section) : isFirstStepValid()
+        }
+        isFirstStep={step === 0}
+        isLastStep={step === sortedSections.length}
+        onNext={() => setCurrentStep(step + 1)}
+        onBack={() => setCurrentStep(step - 1)}
+        isDisabled={
+          isSubmitting ||
+          isValidating ||
+          (step === sortedSections.length &&
+            !shouldNextBtnBeEnabled(sortedSections[step - 1]))
+        }
+        hasAnswers={hasPartialAnswers(sortedSections[step - 1])}
+        isFormInvalid={!isValid}
+        onSave={(review) => {
+          const saveActionValue = review
+            ? SAVE_ACTION_VALUE.SAVE_AND_REVIEW
+            : SAVE_ACTION_VALUE.SAVE;
+
+          methods.setValue(SAVE_ACTION_KEY, saveActionValue);
+
+          methods.handleSubmit(onSubmit, onInvalid)();
+        }}
+        onSaveAsDraft={() => {
+          methods.setValue(SAVE_ACTION_KEY, SAVE_ACTION_VALUE.SAVE_AS_DRAFT);
+          methods.handleSubmit(onSubmit)();
+        }}
+      />
+    );
+  };
+
   return (
     <>
-      {isOpen && (
+      {isOpen && !!sortedSections.length && (
```

Comment:
I don't think we still need this change after updating the sortedSections constant. We should probably add some loading state now that the initial values are async. Maybe we could add a conditional render wrapper within this component.

Reply:
This is actually needed in combination with the update. The update sets sortedSections to [] when loading so that this will clear the Wizard render.

Reply:
Ah I see - noticed that when i remove the useMemo from sortedSections, I no longer need this. Seems like the memoization there is giving us more trouble than it is worth & probably not saving too many re-renders since the assessment prop is cached and managed by react-query. Take it or leave it!